File: backend/pkg/entity/post.go
package entity

import (
	"time"
)

type Posts []Post

const (
	PublicIsPublic              string = "public"
	PrivateIsPublic             string = "private"
	AlmostPrivateIsPublicstring        = "almostPrivate"
)

type Post struct {
	ID         int64     `json:"id"`
	Title      string    `json:"title"`
	Content    string    `json:"content"`
	Image      string    `json:"image"`
	IsPublic   string    `json:"is_public"`
	GroupID    int64    `json:" group_id"`
	UserID     string    `json:"user_id"` // Change this to int64 if you're sending a number
	CreatedAt  time.Time `json:"created_at"`
}

File: frontend/src/entity/post.ts
// export class Post {
//     id!: number
//     title!: string
//     content!: string
//     image?: string;
//     privacy!: string
//     user_id!: number
//     created_at!: Date
//     updated_at!: Date
// }
export class Post {
    id!: number;
    title!: string;
    content!: string;
    image?: string; // Ajoute cette ligne si elle manque
    group_id!: number;
    ispublic!: string;
    user_id!: number;
    created_at!: Date;
}
